File: web-be/src/main/java/com/project/web_be/dtos/AssignExamDTO.java
package com.project.web_be.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class AssignExamDTO {
    @JsonProperty("class_id")
    private Long classId;
    @JsonProperty("exam_id")
    private Long examId;
    @JsonProperty("assigned_date")
    private LocalDateTime assignedDate;
    @JsonProperty("due_date")
    private LocalDateTime dueDate;
}

File: web-fe/src/app/dtos/requests/assign-exam.dto.ts
export class AssignExamDTO {
    assigned_date!: Date;
    due_date!: Date;
    class_id!: number;
    exam_id!: number;

    constructor(data: any) {
        this.class_id = data.class_id;
        this.exam_id = data.exam_id;
        this.assigned_date = data.assignedDate;
        this.due_date = data.dueDate;
    }
}